source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -91,7 +94,12 @@ Config::SharedConfig::SharedConfig(
     max_downstream_connection_duration_ = std::chrono::milliseconds(connection_duration);
   }
 
-  if (config.has_access_log_flush_interval()) {
+  if (config.has_access_log_options() &&
+      config.access_log_options().has_access_log_flush_interval()) {
+    const uint64_t flush_interval = DurationUtil::durationToMilliseconds(
+        config.access_log_options().access_log_flush_interval());
+    access_log_flush_interval_ = std::chrono::milliseconds(flush_interval);
+  } else if (config.has_access_log_flush_interval() /* deprecated */) {
```

Comment:
Please add an check to ensure the new grouped options and deprecated old config will not be configured at same time.

Reply:
What does this check do? Throw an exception? IMO it would be better to have precedence for the grouped options, WDYT?

Reply:
Typically, these fields should never be set at same time. And if it happens, it is an error that need users' additional attention.
IMO, it would be better to throw it out to notice the users.

And I think there are lots of similar handlings in the current envoy code base.

Reply:
done